### DISCUSSION TITLE
Append non-utf8 text to table

### DISCUSSION THREAD

**Discussion by annnei** [duckdb/duckdb#5251] [Q&A] [answered]
I am using duckdb in C++ and the Appender to insert data into a duckdb table.
I have an std::string with a Swedish character Ö inside, and try to append that string with `appender->Append(str.c_str());`
DuckDB does not appreciate that and throws an exception with "Invalid unicode detected in segment statistics update!" (when the appender is being flushed).

How should I handle my string?

The program is running on Windows and the string resides in an older system based on MFC.

**Comment by Mytherin** [comment, marked as answer]
DuckDB only works with UTF8. If your data is encoded in a different encoding you must convert it to UTF8 before handing it over to DuckDB - there is currently no support for alternative (older) encodings.
